src/components/site-search/site-search.client.ts

```diff
@@ -92,8 +99,59 @@ requestAnimationFrame(() => {
 					}
 
 					searchResults.replaceChildren(hdom)
+					resultChildren = searchResults.querySelectorAll('.listitem')
+					resultChildren.forEach((result, index) => {
+						if (index === 0) {
+							result.setAttribute('aria-selected', 'true')
+						} else {
+							result.setAttribute('aria-selected', 'false')
+						}
+					})
 				})
 			})
 		})
+		let indexNumber = 0
+		const onKeydown = (event: { key: string; preventDefault: () => void }) => {
+			if (resultChildren.length > 0) {
+				if (event.key === 'ArrowDown') {
+					event.preventDefault()
+					if (indexNumber < (resultChildren.length - 1)) {
+						indexNumber++
+					}
+					resultChildren.forEach((result: HTMLElement, index: number) => {
+						if (index === indexNumber) {
+							result.setAttribute('aria-selected', 'true')
+						} else {
+							result.setAttribute('aria-selected', 'false')
+						}
```

Comment:
Don’t use this, unless you genuinely prefer it.

```suggestion
						result.setAttribute('aria-selected', index === indexNumber)
```